### PR TITLE
Run Final_Jar_Testing_Linux_GPU in docker

### PR DESCRIPTION
**Pull request by mszhanyi**
### Description
1. Create a package test image based on [RedHat UBI](https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image)
2. Install TensorRT 8.6.1.6 in RedHat. (Ref. https://docs.nvidia.com/deeplearning/tensorrt/install-guide/index.html#maclearn-net-repo-install-rpm)
3. Run Final_Jar_Testing_Linux_GPU in docker (base image: nvidia/cuda:11.8.0-cudnn8-devel-ubi8)

### Motivation and Context
[AB#18470](https://aiinfra.visualstudio.com/6a833879-cd9b-44a4-a9de-adc2d818f13c/_workitems/edit/18470)

### Verification
https://dev.azure.com/aiinfra/Lotus/_build/results?buildId=354004&view=logs&j=8939b564-1402-57b5-92dc-510eba75e069&t=8939b564-1402-57b5-92dc-510eba75e069

### CODE REVIEW

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ENV PATH /usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/src/tensorrt/bin:${PATH}
+
+RUN dnf update -y &&\
+    dnf install -y bash wget &&\
+    dnf clean dbcache
+
+# Install python3
+RUN dnf install -y \
+    python3.8 \
+    python3-pip \
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6:19]: Also change python3-pip and python3-wheel to python3.8-pip and python3.8-wheel.

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    python3-wheel &&\
+    cd /usr/local/bin &&\
+    ln -s /usr/bin/python3 python &&\
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6:22]: /usr/bin/python3 is 3.6. /usr/bin/python3.8 is 3.8

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    ln -s /usr/bin/pip3 pip;
+
+RUN pip install --upgrade pip
+RUN pip install setuptools>=41.0.0
+
+# Install TensorRT
+RUN yum update -y && yum install -y libnvinfer8 libnvonnxparsers8 libnvparsers8 libnvinfer-plugin8 libnvinfer-lean8 libnvinfer-vc-plugin8 libnvinfer-dispatch8
+RUN v="8.6.1.6-1+cuda11.8" &&\
+    yum update -y &&\
+    yum downgrade -y libnvinfer8-${v} libnvinfer8-${v} libnvonnxparsers8-${v} libnvparsers8-${v} libnvinfer-plugin8-${v} libnvinfer-lean8-${v} libnvinfer-vc-plugin8-${v} libnvinfer-dispatch8-${v} &&\
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6:32]: Why downgrade?

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# Licensed under the MIT License.
+# --------------------------------------------------------------
+# Dockerfile to Test ONNX Runtime on UBI8 with CUDA 11.8 and TensorRT 8.6
+
+# Build base image with required system packages
+FROM nvidia/cuda:11.8.0-cudnn8-devel-ubi8 AS base
+
+ENV PATH /usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/src/tensorrt/bin:${PATH}
+
+RUN dnf update -y &&\
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6:12]: As I said in another PR, you do not need to call "dnf update". If here it is apt-get, you should. "apt-get update" is to fetch package metadata. But "dnf upgdate" upgrades packages. It means when you run this command, the cuda packages could be upgraded to a newer version (you haven't call "dnf versionlock" yet).

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 (hunk 3 above)
Comment (mszhanyi) [reply tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6:32]: https://docs.nvidia.com/deeplearning/tensorrt/install-guide/index.html#maclearn-net-repo-install-rpm
Install the TensorRT package that fits your particular needs. When using the NVIDIA CUDA network repository, RHEL will by default install TensorRT for the latest CUDA version. If you need the libraries for other CUDA versions, refer to step 3.
step3: The following commands install libnvinfer8 and related TensorRT packages for an older CUDA version and hold the libnvinfer8 package at this version. Replace 8.x.x.x with your version of TensorRT and cudax.x with your CUDA version for your install.

Comment (snnn) [reply tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6:32]: Thanks for the explanation

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 (hunk 4 above)
Comment (mszhanyi) [reply tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6]: thx, updated!

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    python3-wheel &&\
+    cd /usr/local/bin &&\
+    ln -s /usr/bin/python3 python &&\
+    ln -s /usr/bin/pip3 pip;
+
+RUN pip install --upgrade pip
+RUN pip install setuptools>=41.0.0
+
+# Install TensorRT
+RUN yum update -y && yum install -y libnvinfer8 libnvonnxparsers8 libnvparsers8 libnvinfer-plugin8 libnvinfer-lean8 libnvinfer-vc-plugin8 libnvinfer-dispatch8
```
[the remark this answers is not in the document]
Comment (mszhanyi) [reply tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6]: updated!

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+# Install python3
+RUN dnf install -y \
+    python3.8 \
+    python3-pip \
+    python3-wheel &&\
+    cd /usr/local/bin &&\
+    ln -s /usr/bin/python3 python &&\
+    ln -s /usr/bin/pip3 pip;
+
+RUN pip install --upgrade pip
```
[the remark this answers is not in the document]
Comment (mszhanyi) [reply tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6]: updated!

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 (hunk 2 above)
Comment (mszhanyi) [reply tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6]: updated!

File: tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6 (hunk 1 above)
Comment (mszhanyi) [reply tools/ci_build/github/linux/docker/Dockerfile.package_ubi8_cuda11_8_tensorrt8_6]: thks, updated!

**Review by snnn — DISMISSED** [review]
Thank you!

**Review by snnn — APPROVED** [review]

**Pull request merged**